napari/_qt/qt_viewer.py

```diff
@@ -1174,7 +1174,9 @@ def closeEvent(self, event):
         event : qtpy.QtCore.QCloseEvent
             Event from the Qt context.
         """
-        self.layers.close()
+        if self._layers is not None:
+            # do not create layerlist if it does not exist yet.
+            self.layers.close()
```

Comment:
Is this to say that before, even if a layerlist hadn't been created, by accessing `self.layers`, we were creating one just to then close it?

Reply:
Yes